cmake/Modules/FindMKL.cmake

```diff
@@ -212,7 +216,11 @@ MACRO(CHECK_ALL_LIBRARIES LIBRARIES OPENMP_TYPE OPENMP_LIBRARY _name _list _flag
         # Separately handling compiled TBB
         SET(_found_tbb TRUE)
       ELSE()
-        FIND_LIBRARY(${_prefix}_${_library}_LIBRARY NAMES ${_library})
+        SET(lib_names ${_library})
+        IF (BUILD_SHARED_LIBS)
```

Comment:
Actually we are using the static version of MKL even when we build the dynamic version(`BUILD_SHARED_LIBS=ON`). So we may need to create a new variable here.